lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php

```diff
@@ -187,6 +187,16 @@ public function loadMetadataForClass(
                 $associationMapping['fieldName']    = $this->getFieldNameForColumn($manyTable->getName(), current($otherFk->getColumns()), true);
                 $associationMapping['targetEntity'] = $this->getClassNameForTable($otherFk->getForeignTableName());
 
+		if (isset($metadata->fieldMappings[$associationMapping['fieldName']])
+                        || isset($metadata->associationMappings[$associationMapping['fieldName']])) {
+                        $ii=2;
+                        while ( isset($metadata->fieldMappings[$associationMapping['fieldName'].(string)$ii]) ||
+                                isset($metadata->associationMappings[$associationMapping['fieldName'].(string)$ii])) {
```

Comment:
This maps multiple associations over the same join columns: this is outside what ORM can handle and is designed for

Reply:
Hello,
Thanks for answering so quickly @Ocramius 
Though I'm not sure to understand properly the "corrupting ORM invariants about valid association mappings" part... So I'd be glad if you could detail your analysis for me tobetter understand.
Because my first goal is to properly generate correct entities from an old database and If my patch creates corrupt PHP code that's a big problem and I need to understand the consequences (and more important how to fix this corruption problem)
My second goal, as a bonus, is to provide  a mean for others, who will meet these use cases, to work with Doctrine without problem : If I can solve it for mayself that could be great to solve it for all